### PR TITLE
Vsebp 52 as a user i want to be able to see detailed information for a specific event such that i can have an overview of that event

### PR DESCRIPTION
**Pull request by Kasper-S-Jensen**
- implement frontend for multiple user stories including:
See detailed event information
Create event
See all current events

all user stories are complete with validation and tests

### CODE REVIEW

File: app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/data/repository/EventRepository.kt [hunk 1]
```diff
@@ -1,42 +1,164 @@
...
+import io.github.viabachelora23michaelkutaibakasper.bprapp.data.domain.User
+import io.github.viabachelora23michaelkutaibakasper.bprapp.type.GeoLocationInput
+import io.github.viabachelora23michaelkutaibakasper.bprapp.type.UserInput
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
 
 class EventRepository : IEventRepository {
+    fun parseUtcStringToLocalDateTime(utcString: String): LocalDateTime {
```
Comment (michaelbui99) [inline app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/data/repository/EventRepository.kt:22]: Move to util or make extension method

File: app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/ui/screens/events/createevent/TitleAndDescriptionScreen.kt [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            modifier = Modifier
+                .fillMaxWidth()
+        )
+        {
+
+
+            // Save or submit button
+            Button(
+                onClick = {
+                    if (isInvalidTitle(title) || isInvalidDescription(description)) {
```
Comment (michaelbui99) [inline app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/ui/screens/events/createevent/TitleAndDescriptionScreen.kt:96]: Use properties from viewmodel

**Review by michaelbui99 — CHANGES_REQUESTED** [review]
Noice 🚀

File: app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/data/repository/EventRepository.kt (hunk 1 above)
Comment (Kasper-S-Jensen) [reply app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/data/repository/EventRepository.kt:22]: done

File: app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/ui/screens/events/MapView.kt [hunk 3]
```diff
@@ -15,114 +16,122 @@ import androidx.compose.foundation.lazy.LazyColumn
...
 
-            when (selectedIndex.intValue) {
-                mapView -> {
-                    mapEvents()
-                }
+        when (selectedIndex.intValue) {
+            mapView -> {
+                MapEvents(
+                    navController = navController,
+                    LatLng(55.862207, 9.844651),
```
[the remark this answers is not in the document]
Comment (Kasper-S-Jensen) [reply app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/ui/screens/events/MapView.kt:95]: done

**Review by Kutaiba-Nezar-Kashmar — APPROVED** [review]
🚀🚀 nicely done !!

**Review by michaelbui99 — APPROVED** [review]

**Pull request merged**